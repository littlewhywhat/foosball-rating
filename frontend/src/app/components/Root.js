import React from 'react'
import { withRouter, Route } from 'react-router-dom'
import { Header } from './header'
import { Footer } from './Footer/footer'
import { Container } from '../../styles/blocks/layout'
import { Game } from './Game'
import { SelectGameByUrl } from './SelectGameByUrl'
import { ShowIfGameIsSelected } from './ShowIfGameSelected'
import { SelectGamePage } from '../pages/SelectGamePage'
import { GameNotFoundPage } from '../pages/GameNotFound'

class RootComponent extends React.Component {
  render() {
    return <>
      <Route path='/:gameName' component={SelectGameByUrl}/>
      <Header/>
      <Container>
        <ShowIfGameIsSelected>
          <Route path='/:gameName' component={Game} />
          <Route path='/:gameName' component={GameNotFoundPage}/>
        </ShowIfGameIsSelected>
        <Route exact path='/' component={SelectGamePage}/>
      </Container>
      <Footer/>
    </>
  }
}

export const Root = withRouter(RootComponent)
