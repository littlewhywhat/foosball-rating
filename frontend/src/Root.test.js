import React from 'react'
import ReactDOM from 'react-dom'
import { Root } from './Root'

import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Root />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('matches snapshot', () => {
  expect(shallow(<Root/>)).toMatchSnapshot()
})
