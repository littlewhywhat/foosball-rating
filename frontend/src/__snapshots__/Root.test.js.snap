// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ThemeComponent)>
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          path="/foosball"
        />
        <Route>
          <Redirect
            to="/foosball"
          />
        </Route>
      </Switch>
    </BrowserRouter>
  </Connect(ThemeComponent)>
</Provider>
`;
